tests/test_utils.py

```diff
@@ -174,421 +282,128 @@ def test_parse_log_components(ns_3_compiled_debug, config):
-         'Severity_class': 'FUNCTION',
-         'Message': ''
-         },
-        {'Time': 0.045510017,
-         'Context': '1',
-         'Extended_context': 'mac=00:00:00:00:00:01',
-         'Component': 'FrameExchangeManager',
-         'Function': 'RxStartIndication',
-         'Arguments': '0x5576595683e0, "PSDU reception started for ", +76us, " (txVector: ", txpwrlvl: 17 preamble: LONG channel width: 20 GI: 800 NTx: 97 Ness: 0 MPDU aggregation: 0 STBC: 0 FEC coding: BCC mode: OfdmRate6Mbps Nss: 1, ")"',
-         'Severity_class': 'FUNCTION',
-         'Message': ''
-         },
-        {'Time': 2.999074264,
-         'Context': '0',
-         'Extended_context': 'mac=00:00:00:00:00:02',
-         'Component': 'FrameExchangeManager',
-         'Function': 'NotifyReceivedNormalAck',
-         'Arguments': '0x56239764d8c0, DATA, payloadSize=1456, to=00:00:00:00:00:01, seqN=2323, duration/ID=+44000ns, lifetime=+195539us, packet=0x56239786dd00',
-         'Severity_class': 'FUNCTION',
-         'Message': ''
-         },
-        {'Time': 0.0,
-         'Context': '-1',
-         'Extended_context': None,
-         'Component': 'WifiPhy',
-         'Function': 'WifiPhy',
-         'Arguments': '0x557659571bd0',
-         'Severity_class': 'FUNCTION',
-         'Message': ''
-         },
-        {'Time': 0.000000000,
-         'Context': '-1',
-         'Extended_context': None,
-         'Component': 'WifiPhy',
-         'Function': 'SetChannelNumber',
-         'Arguments': '',
-         'Severity_class': 'DEBUG',
-         'Message': 'Saving channel number configuration for initialization'
-         },
-        {'Time': 0.000000000,
-         'Context': '-1',
-         'Extended_context': None,
-         'Component': 'WifiPhy',
-         'Function': 'SetChannelNumber',
-         'Arguments': '',
-         'Severity_class': 'INFO',
-         'Message': '()[]./-='
-         },
-        {'Time': 2.999825333,
-         'Context': '1',
-         'Extended_context': None,
-         'Component': 'WifiPhy',
-         'Function': 'GetTxPowerForTransmission',
-         'Arguments': '',
-         'Severity_class': 'DEBUG',
-         'Message': 'txPowerDbm=17 with txPowerDbmPerMhz=3.9897 over 20 MHz'
-         },
-        {'Time': 2.999825333,
-         'Context': '1',
-         'Extended_context': None,
-         'Component': 'WifiPhy',
-         'Function': 'GetTxPowerForTransmission',
-         'Arguments': '',
-         'Severity_class': 'INFO',
-         'Message': 'txPowerDbm=17 after applying m_powerDensityLimit=100'
-         }
-    ]
-    wipe_results(db, db_dir)
-    os.remove(data_dir)
+    try:
+        db = TinyDB(db_dir,
+                    storage=CachingMiddleware(JSONStorage))
+
+        with open(data_dir, "w") as f:
+            f.writelines(log_ls)
+
+        parse_list = parse_logs(data_dir)
+        insert_logs(parse_list, db)
+
+        # Test case 1
+        filter_list = filter_logs(db, context=['1'])
```

Comment:
you're always inserting lists of strings as arguments, have you checked the other possibilities?
I.e., string, number, list of numbers
Tests should also catch that erratic behavior, and it should be done for all input arguments (you never know if a typo or a future distraction broke just one of them)

Reply:
Added in [1d87256](https://github.com/signetlabdei/sem/pull/51/commits/1d87256a25d9aff5dab4c1b31badbcb2f6bd67f4).